lib/src/op_store.rs

```diff
@@ -261,6 +261,7 @@ pub struct View {
     // (.jj/working_copy/) has the source of truth about which commit *is* checked out (to be
     // precise: the commit to which we most recently completed an update to).
     pub wc_commit_ids: BTreeMap<WorkspaceNameBuf, CommitId>,
+    pub workspace_roots: BTreeMap<WorkspaceNameBuf, Vec<u8>>,
```

Comment:
Maybe better to manage `{commit_id: CommitId, root: PathBuf}` in a single mapping? That will help keep them sync. We'll have to delete the root path entry when concurrent operations are merged and the workspace gets deleted, for example.

Another idea is to store root paths as completely separate auxiliary data. We could add a new proto file that just keeps track of `workspace_roots` mapping. In this model, the paths mapping may be out of sync on e.g. `jj undo` (just like git remotes.)

@martinvonz might have a better idea.

Reply:
Hmm.. I don't understand too many internals, so can you please let me know where exactly should I store this?

Reply:
I am asking because @yuja proposed two different suggestions. I would be okay if you guys decide, and let me know where.

Reply:
My comment about the data type was just to highlight that it's unclear what it should be if it's part of the data model. I agree that it's better to not put it in the view object. I also agree that it feels like there's some possible additional abstraction around `WorkspaceLoader`/`WorkspaceLoaderFactory`.

Reply:
Pushed a draft, I was able to add a `WorkspaceLoaderTable` and use it in commands, but unfortunately I couldn't figure out how to persist it. Therefore, I implemented a naive config read/update too.

Would appreciate some guidance.

Reply:
@martinvonz Can I get your thoughts on the current approach? I would love to finish this PR.